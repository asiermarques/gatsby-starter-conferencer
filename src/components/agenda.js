import React from "react"
import { graphql, StaticQuery } from "gatsby"
import agendaMapper from "../lib/agendaMapper"
import agendaSlotType from "../lib/agendaSlotTypes";

export const PureAgenda = ({ data }) => {
            
    const tracks = data.site.siteMetadata.agenda.tracks;
    const agenda = agendaMapper(data.site.siteMetadata.agenda, data.site.siteMetadata.speakers);
    
    return (
    <section id="agenda">
        <div class="container">
            <h2 class="text-center">Agenda</h2>
            <div class="table-responsive-sm">
            <table id="agenda-table" class="table table-striped stacktable large-only">
                    <thead>
                    <tr>
                        <th></th>
                        {tracks.map( (track, index) => <th key={index} width="40%">{track.name}</th> )}
                    </tr>
                    </thead>
                    <tbody>
                    {agenda.map( (slot, slot_index) => { 
                        const [time_range, ...content] = slot;
                        return (<tr key={slot_index}>
                            <td class="time">{time_range}</td>
                            {content.map( (content_item, index) => {
                                
                                if(content_item.type === agendaSlotType.SPEAKER) { 
                                    return (
                                        <td key={index}>
                                            <a href={content_item.content.slug + "#talk"}>
                                                <h5>{content_item.content.talk.title}</h5>
                                                <p>{content_item.content.name}</p>
                                            </a>
                                        </td>
                                    )
                                }else{
                                    return (<td key={index}><p>{content_item.content}</p></td>)
                                }
                            })}
                        </tr>
                        )
                    })}
                    </tbody>
            </table>
            </div>
        </div>
    </section>
)}

export const Agenda = props => (
    <StaticQuery
        query={graphql`
        query {
            site {
                siteMetadata {
                    speakers {
                        slug,
                        name,
                        company,
                        talk{
                            title,
                            description
                        }
                    },
                    agenda {
                        visible,
                        time_slots,
                        tracks{ 
                            name,
                            content_in_slots {
                                type,
                                content
                            }
                        }
                    }
                }
            }
        }
        `}
        render={data => <PureAgenda {...props} data={data} />}
    />
)