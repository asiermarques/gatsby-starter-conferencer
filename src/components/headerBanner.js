import React from "react"
import { graphql, StaticQuery } from "gatsby"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
                siteMetadata {
                    conference_name, 
                    conference_claim,
                    header_banner {
                        cta_pre_text,
                        cta_text
                    }
                }
            }
        }
        `}
        render={ data => (
            <div class="jumbotron">
                <div class="container">
                    <h1 class="display-4">{data.site.siteMetadata.conference_name}</h1>
                    <p class="lead">{data.site.siteMetadata.conference_claim}</p>
                    <hr class="my-4"/>
                    <p>{data.site.siteMetadata.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href="/dot" role="button">{data.site.siteMetadata.header_banner.cta_text}</a>
                </div>
            </div>
        )}
    />
)