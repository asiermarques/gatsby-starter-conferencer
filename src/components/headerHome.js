import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container text-center">
            <h1 class="display-4">{data.conference_name}</h1>
            <p class="lead">{data.conference_claim}</p>
            <hr class="my-4"/>
            <p>{data.header_banner.cta_pre_text}</p>
            <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
        </div>
    </div>