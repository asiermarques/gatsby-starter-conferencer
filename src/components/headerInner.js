import React from "react"

export default ({ data }) => <div class="jumbotron">
        <div class="container">
            <div className="row">
                <div className="col-sm-7">
                    <a href="/"><h2 class="display-4">{data.conference_name}</h2></a>
                    <p class="lead">{data.conference_claim}</p>
                </div>
                <div className="col-sm-5 text-center">
                    <p>{data.header_banner.cta_pre_text}</p>
                    <a class="btn btn-primary btn-lg" href={data.header_banner.cta_url} role="button">{data.header_banner.cta_text}</a>
                </div>
            </div> 
        </div>
    </div>
  

