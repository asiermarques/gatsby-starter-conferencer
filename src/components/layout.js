import React from "react"
// eslint-disable-next-line
import Styles from "../scss/custom.scss"

export default ({ children }) => (
  <div>
    {children}
  </div>
)