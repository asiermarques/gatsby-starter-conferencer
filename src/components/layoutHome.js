import React from "react"
import Styles from "../scss/custom.scss"

import Header from "./headerBanner"

export default ({ children }) => (
  <div>
    <Header></Header>
    {children}
  </div>
)

