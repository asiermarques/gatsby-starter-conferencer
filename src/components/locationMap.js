import React from "react"

export default ({ location }) => <section id="where">
        <h2 class="text-center">Location</h2>
        <div class="content_wrap">

          <div class="content">
            <div class="container">
              <div class="col-md-5 col-xs-12">

                <div class="card">
                  <div class="card-body">
                  {location.addresses.map( (address, index) => (<div key={index} class="address">

                    <h5>{address.name}</h5>
                    <p> 
                      <span dangerouslySetInnerHTML={{__html:address.line}}></span>
                      <br></br>
                      <small><a target="_blank" rel="noopener noreferrer" href={address.map_link}>open map</a></small>
                    </p>

                  </div>) )}
                  </div>
                </div>

              </div>
            </div>
          </div>

          <iframe src={location.gmaps_iframe_url}
          width="100%" 
          height="550" 
          title="map"
          frameborder="0" 
          allowfullscreen="false"></iframe>
        </div>
    </section>