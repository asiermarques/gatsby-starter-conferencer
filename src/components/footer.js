import React from "react"
import { graphql, StaticQuery } from "gatsby"
import {LocationMap} from "./locationMap"
import {Sponsors} from "./sponsors"
import Organizers from "./organizers"
import Nav from "./nav"


export default () => (
    <StaticQuery
        query={graphql`
        query {
            site {
              siteMetadata {
                conference_hashtag
                organizers {
                  name,
                  link,
                  image
                },
                footer_links {
                  link,
                  title
                }
              }
            }
          }
        `}
        render={ data => (
    <footer>
        <LocationMap/>

        <section id="about">

          <div class="container">

              <div class="row">
                    <div class="col-md-7">
                      <Organizers items={data.site.siteMetadata.organizers}/>
                    </div>

                    <div class="col-md-5 text-center">
                      <h4>About</h4>
                      <Nav items={data.site.siteMetadata.footer_links}></Nav>
                    </div>

              </div>
          </div>

        </section>

        <Sponsors/>

        <section>
            <div className="container">
                <hr></hr>
                Created with the <a href="https://github.com/asiermarques/gatsby-starter-conferencer">Gatsby Conferencer starter</a>
            </div>
        </section>

                

    </footer>
        )}
    />
)