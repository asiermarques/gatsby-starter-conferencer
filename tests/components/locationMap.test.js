import React from "react"
import renderer from "react-test-renderer"
import {PureLocationMap as LocationMap} from "../../src/components/locationMap"
import LocationConfigFixture from "../__fixtures/LocationConfig"

describe("LocationMap", () => {
  it("renders correctly", () => {
    const data = {
        site: {
          siteMetadata: {
            "location": LocationConfigFixture
          },
        },
      }
    const tree = renderer.create(<LocationMap data={data}/>).toJSON()
    expect(tree).toMatchSnapshot()
  });
})