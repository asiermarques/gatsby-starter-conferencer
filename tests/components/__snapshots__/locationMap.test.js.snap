// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationMap renders correctly 1`] = `
<section
  id="where"
>
  <h2
    class="text-center"
  >
    Location
  </h2>
  <div
    class="content_wrap"
  >
    <div
      class="content"
    >
      <div
        class="container"
      >
        <div
          class="col-md-5 col-xs-12"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <div
                class="address"
              >
                <h5>
                  Main Venue
                </h5>
                <p>
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "Example Street 140<br>Gotham 48480",
                      }
                    }
                  />
                  <br />
                  <small>
                    <a
                      href="https://www.google.com/maps?ll=43.266707,-2.945395&z=17&t=m&hl=es&gl=ES&mapclient=embed"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      open map
                    </a>
                  </small>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <iframe
      allowfullscreen="false"
      frameborder="0"
      height="550"
      src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2905.255497835498!2d-2.947592084396059!3d43.26702288533392!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x0!2zNDPCsDE2JzAxLjMiTiAywrA1Nic0My41Ilc!5e0!3m2!1ses!2ses!4v1543099024128"
      title="map"
      width="100%"
    />
  </div>
</section>
`;
